rama-dns/src/lib.rs

```diff
@@ -79,6 +80,37 @@ impl<R: DnsResolver<Error: Into<BoxError>>> DnsResolver for Option<R> {
     }
 }
 
+static_str_error! {
+    #[doc = "domain not mapped in any dns resolver"]
+    pub struct DomainNotFoundErr;
+}
+
+impl<R: DnsResolver> DnsResolver for Vec<R> {
```

Comment:
The issue requested the implementation for `Vec` and also `Array`.

If you only have bandwidth to do `Vec` that's okay, but know the array is pretty much the same.

The signature to impl Arrays would be something like:

```
impl<R: DnsResolver, const N: usize> DnsResolver for [R; N] {
    // ...
}
```

Something like that. Typing directly in comment here, so might have gotten some syntax wrong, but it's more or less like that